### PR TITLE
fix: correct Long types import

### LINKED ISSUE #345: Missing explicit dependency on Long.js
The `datastore.d.ts` and `entity.d.ts` files that are included in the build uses Long.js. There are several type definitions there that use the `Long` type.

Long.js is not declared in package.json as a direct dependency of  `nodejs-datastore`, which causes compilation to fail in some monorepo situations where dependencies may get hoisted. In our case, using yarn.

It works as expected without monorepos, because long.js is a transitive dependency from protobuf.js

  - `@google-cloud/datastore` version: 3.1.1

#### Steps to reproduce

Sorry, wasn't able to reproduce outside of our monorepo, which I'm not able to share.

Cheers

### PR DESCRIPTION
**Pull request by SimenB**
Fixes #345

Ref #352

### CODE REVIEW

**Comment by SimenB** [comment]
ergh no, the generator emoves the manual edit I made 😅 Does need the dependency in package.json, though

File: package.json [hunk 1]
```diff
@@ -49,6 +49,7 @@
     "@google-cloud/projectify": "^0.3.2",
     "@google-cloud/promisify": "^0.4.0",
     "@types/duplexify": "^3.6.0",
+    "@types/long": "^4.0.0",
```
Comment (JustinBeckwith) [inline package.json]: you don't need this because it's shipped with `protobufjs` :)

File: proto/datastore.d.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import * as $protobuf from "protobufjs";
-import * as long from "long";
+import * as Long from "long";
```
Comment (JustinBeckwith) [inline proto/datastore.d.ts]: Yarrr, nice find and sorry for the mixup.  What you *really* wanna do here is update this line in package.json:

```
"proto:datastore": "mkdir -p proto && pbjs -t static-module -w commonjs -p node_modules/google-proto-files google/datastore/v1/datastore.proto | pbts -i long -o proto/datastore.d.ts -",
```

Make that:
```
"proto:datastore": "mkdir -p proto && pbjs -t static-module -w commonjs -p node_modules/google-proto-files google/datastore/v1/datastore.proto | pbts -i Long -o proto/datastore.d.ts -",
```

That way the fix doesn't get roughed up by the generator.

Comment (SimenB) [reply proto/datastore.d.ts]: Then it's `import Long from 'Long'` which I guess will go boom on linux?

File: package.json (hunk 1 above)
Comment (SimenB) [reply package.json]: Yeah, but you shouldn't depend on it existing transitively. It won't work with `yarn pnp` for instance

File: proto/datastore.d.ts (hunk 2 above)
Comment (JustinBeckwith) [reply proto/datastore.d.ts]: Nah, it'll probably be fine

File: package.json (hunk 1 above)
Comment (JustinBeckwith) [reply package.json]: ah, cool - did not know that!

File: proto/datastore.d.ts (hunk 2 above)
Comment (mgabeler-lee-6rs) [reply proto/datastore.d.ts]: This very much will blow up on Linux -- import filenames are case sensitive there.  Applying the suggestion results in:
```
> @google-cloud/datastore@3.1.2 compile /home/mgl/3psrc/nodejs-datastore
> tsc -p . && cp -r src/v1 build/src && cp -r proto* build && cp test/*.js build/test

proto/datastore.d.ts:2:23 - error TS2307: Cannot find module 'Long'.

2 import * as Long from "Long";
                        ~~~~~~


Found 2 errors.
```

Comment (stephenplusplus) [reply proto/datastore.d.ts]: Is this what we want? https://github.com/dcodeIO/protobuf.js/pull/1166

Comment (JustinBeckwith) [reply proto/datastore.d.ts]: Yes lol.  That would fix all the things.

**Comment by callmehiphop** [comment]
@JustinBeckwith in some of the other clients we are just editing the generated file until a better solution presents itself. Assuming we're cool doing that here as well, I think this PR is good to go unless you have any objections

**Comment by JustinBeckwith** [comment]
SGTM.  Let's just be careful about not blasting these away :)

**Comment by callmehiphop** [comment]
@JustinBeckwith I see you became a contributor to Protobuf.js 😮 any chance we'll see a release soon with the Long fix?

**Pull request merged**